src/api/data/charts.go

```diff
@@ -8,32 +8,47 @@ import (
 	"github.com/helm/monocular/src/api/pkg/swagger/models"
 )
 
-// GetChart gets the chart associated with the passed-in repo+name
-func GetChart(repo, name string) (models.Resource, error) {
-	// TODO implement actual "get chart" business logic
+// Charts is an interface for managing chart data
+type Charts interface {
+	// will have a GetChart method to retrieve a particular chart from a repo
+	GetChart(repo, name string) (models.Resource, error)
+	// will have a GetAllFromRepo method to retrieve all charts from a repo
+	GetAllFromRepo(repo string) ([]*models.Resource, error)
+	// will have a GetAll method to retrieve all charts from all repos
+	GetAll() ([]*models.Resource, error)
+}
+
+// mockCharts fulfills the Charts interface
+type mockCharts struct{}
+
+// NewMockCharts returns a new mockCharts
+func NewMockCharts() Charts {
+	return &mockCharts{}
+}
+
+// GetChart method for mockCharts
+func (g *mockCharts) GetChart(repo, name string) (models.Resource, error) {
```

Comment:
There are a lot of functions that look like this, but this looks sorta suspect to me. Basically, you match a method signature, run a function with a different name, trap its error, hide the error with a replacement error, and then return the results. It's impossible to recover the actual cause of any given error. This seems really convoluted. Seems to me it could have just been reduced to `return mocks.GetChartFromMockRepo(repo, name)`